pkg/controller/controller.go

```diff
@@ -1162,7 +1165,7 @@ func (p *csiProvisioner) Delete(ctx context.Context, volume *v1.PersistentVolume
 			klog.Warningf("failed to get storageclass: %s, proceeding to delete without secrets. %v", storageClassName, err)
 		}
 	}
-	deleteCtx, cancel := context.WithTimeout(ctx, p.timeout)
+	deleteCtx, cancel := context.WithTimeout(markAsMigrated(ctx, hasMigrated(p, storageClass)), p.timeout)
```

Comment:
Should we add a boolean here to record if the PV is actually an in-tree PV
https://github.com/kubernetes-csi/external-provisioner/blob/aef465af4d2649eda1b8e63c9f8db9be257b27b5/pkg/controller/controller.go#L1101

And we can just pass the bool here?

Reply:
Do you mean that we could set a bool value here:https://github.com/kubernetes-csi/external-provisioner/blob/aef465af4d2649eda1b8e63c9f8db9be257b27b5/pkg/controller/controller.go#L1101, something like 
https://github.com/kubernetes-csi/external-provisioner/blob/aef465af4d2649eda1b8e63c9f8db9be257b27b5/pkg/controller/controller.go#L517 and then there's no need to have the hasMigrated method?

Reply:
Yep